### DISCUSSION TITLE
How to make renovate know that given maven property is version of an artifact it should consider?

### DISCUSSION THREAD

**Discussion by marcinpz** [renovatebot/renovate#11104] [Request Help] [answered]
I have in my `pom.xml`:
```xml
...
<properties>
  <mq.version>1.0</mq.version>
</properties>
...
<plugin>
  <groupId>org.codehaus.mojo</groupId>
  <artifactId>exec-maven-plugin</artifactId>
  <executions>
      <execution>
        <configuration>
          <executable>script.sh</executable>
          <environmentVariables>
              <MQ_VERSION>${mq.version}</MQ_VERSION>
          </environmentVariables>
      </configuration>
  </execution>
</plugin>
```
and `script.sh`:
```sh
docker-compose up
```
and `docker-compose.yaml`:
```
services:
  mq:
    image: mq:${MQ_VERSION}
```

How to make Renovate updated `mq.version` property in `pom.xml` when new version of `mq` arrives?
I see properties are supported by Renovate as it works with `jar` dependencies. 
But it does not work when parameter is passed as env variable to shell script and then to docker-compose config file.

**Comment by viceice** [comment, marked as answer]
Where is the `MQ_VERSION` var defined? you can use the regex manager to update arbitrary text files.

Checkout our preset for multiple examples: https://github.com/renovatebot/.github/blob/main/default.json#L97

**Reply by marcinpz** [reply]
Thank you @viceice 
`MQ_VERSION` is defined in `environmentVariables` in `pom.xml` and passed to the `script.sh` with `exec-maven-plugin`. So replacing `MQ_VERSION` is not the solution here. I need `mq.version` param to be updated because I need to be able to overwrite version with maven:
```
mvn -Dmq.version=1.2
```

**Reply by viceice** [reply]
can you create a minimal reproduction repo, so we can workaout a solution for you

https://github.com/renovatebot/renovate/blob/b30f9b8b3e53deac242190fe1091ab85c07df976/docs/development/minimal-reproductions.md

**Reply by marcinpz** [reply]
Here it is: https://github.com/marcinpz/renovate-update-maven-property-poc

**Reply by viceice** [reply]
So it looks like you need a regex manager to match `<mq.version>3.8</mq.version>` as a dependency.

See our preset for samples: https://github.com/renovatebot/.github/blob/main/default.json#L97

**Reply by marcinpz** [reply]
Thank you @viceice!!!
Something like this works:
```json
{  
"regexManagers": [
    {
      "fileMatch": [
        "(^|/)pom.xml$"
      ],
      "matchStrings": [
        "<mq.version>(?<currentValue>.*)</mq.version>"
      ],
      "depNameTemplate": "rabbitmq",
      "datasourceTemplate": "docker",
      "versioningTemplate": "docker"
    }
]
}
```

**Reply by viceice** [reply]
You should escape the dot and the slash with a double backsplash, otherwise the regex can errors out of do a wrong match.
